src/Aspire.Hosting.Azure.CognitiveServices/AzureOpenAIExtensions.cs

```diff
@@ -76,6 +90,20 @@ public static IResourceBuilder<AzureOpenAIResource> AddAzureOpenAI(this IDistrib
                       .WithParameter(AzureBicepResource.KnownParameters.PrincipalId)
                       .WithParameter(AzureBicepResource.KnownParameters.PrincipalType)
                       .WithManifestPublishingCallback(resource.WriteToManifest);
+
+        static void AddDependency(global::Azure.Provisioning.Resource resource, global::Azure.Provisioning.Resource dependency)
+        {
+            // abstract class Resource
+            // { 
+            //     internal void AddDependency(Resource resource)
+            //     {
+            //         Dependencies.Add(resource);
+            //     }
+            // }
+
+            var addDependencyMethod = typeof(global::Azure.Provisioning.Resource).GetMethod("AddDependency", BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
Do we want to cache this? 

Or just wait until this method is public I guess....

Reply:
I didn't think it would matter since it's called on provisioning only, and in rare cases.